cmd/livepeer/starter/starter.go

```diff
@@ -737,16 +746,26 @@ func StartLivepeer(ctx context.Context, cfg LivepeerConfig) {
 				// Can't divide by 0
 				panic(fmt.Errorf("-pixelsPerUnit must be > 0, provided %d", *cfg.PixelsPerUnit))
 			}
+			pixelsPerUnit := new(big.Rat).SetInt64(int64(*cfg.PixelsPerUnit))
 			if cfg.PricePerUnit == nil {
 				// Prevent orchestrators from unknowingly providing free transcoding
 				panic(fmt.Errorf("-pricePerUnit must be set"))
 			}
-			if *cfg.PricePerUnit < 0 {
-				panic(fmt.Errorf("-pricePerUnit must be >= 0, provided %d", *cfg.PricePerUnit))
+			pricePerUnit, currency, err := parsePricePerUnit(*cfg.PricePerUnit)
```

Comment:
I think you'll also need to add parsing here: https://github.com/livepeer/go-livepeer/blob/a3e725c956ad271dc6b0aeda6025053bd4a6f72f/server/handlers.go#L124

That makes me think that maybe we should store the "unparsed" price in Broadcaster in `BroadcastConfg.maxPrice` and only later convert. Think about the following scenario:
1. Someone starts Broadcaster with `-maxPricePerUnit=500USD`.
2. The `startPriceUpdateLoop` is started
3. Someone changes the Broadcaster maxPricePerUnit to `300`
4. The loop `startPriceUpdateLoop` should not be running anymore

The same in the opposite situation.

Reply:
Ok, that was a big miss in this PR. Spent quite some time thinking how I could abstract this on top of the current architecture and had to refactor a lot of stuff. I've proposed it in the commit https://github.com/livepeer/go-livepeer/pull/2981/commits/12060d7e6399cdcdb12e29618fe9fb7604034797 (sorry included some minor review fixes by accident there). Notice that I haven't updated the tests yet, but will do if we align on the direction there.

None of the ideas I could think of for this seemed ideal. I didn't want to add a lot of logic on the "pure config" components like `BroadcasterCfg`, while at the same time didn't want to do a bigger refactor and remove the pricing info from them. In this solution I went with, I don't like that `onUpdate` part, which also causes repetition of the logs and monitor code. At the same time it felt like an auxiliary/non-complex thing, so not worth abstracting further. 

One thing to consider is if it'd be better to just remove the prices from the "dumb" config components and just move them to a `core.TranscodingPriceProvider` or sth like that which gets used instead. This might incur a bigger refactor, but could be simpler to reason about and maintain. Let me know if you'd prefer that or any other idea you have.